cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -195,15 +271,41 @@ impl BuildEnvNix {
     /// this function is currently assumes that the package is from the nixpkgs base-catalog.
     /// Currently the type is distinguished by the [LockedPackageCatalog::locked_url].
     /// If this does not indicate a nixpkgs package, the function will currently panic!
-    fn realise_nixpkgs(&self, locked: &LockedPackageCatalog) -> Result<(), BuildEnvError> {
-        // check if all store paths are valid, or can be substituted
-        // if so, return without eval
+    fn realise_nixpkgs(
+        &self,
+        client: &impl ClientTrait,
+        locked: &LockedPackageCatalog,
+    ) -> Result<(), BuildEnvError> {
+        // Check if all store paths are valid, or can be substituted.
         let all_valid = self.check_store_path_with_substituters(locked.outputs.values())?;
 
+        // If so, return without eval.
         if all_valid {
             return Ok(());
         }
 
+        let installable = {
+            let mut locked_url = locked.locked_url.to_string();
+
+            if let Some(revision_suffix) = locked_url.strip_prefix(NIXPKGS_CATALOG_URL_PREFIX) {
+                locked_url = format!("{FLOX_NIXPKGS_PROXY_FLAKE_REF_BASE}/{revision_suffix}");
+            } else {
+                debug!(?locked.attr_path, "Trying to substitute custom package");
+                let all_found = self.try_substitute_custom_pkg(client, locked)?;
+                // We asked for all the outputs for the package, got store info for
+                // each, and were able to substitute them all.  If so, then we're done here.
+                if all_found {
+                    return Ok(());
+                };
+                panic!("Building custom packages is not yet supported");
```

Comment:
nb: would still prefer `todo!` as it cathces the intend better but never mind if that conflicted with asserting in the test.

Reply:
Will consider a follow up, thanks.